weather-station/background/main.js

```diff
@@ -0,0 +1,358 @@
+let Wappsto = require("wapp-api");
+let axios = require("axios");
+let config = require("./config");
+let networkInfo = require("./networkInfo.json");
+
+let wappsto = new Wappsto({
+  baseUrl: "https://wappsto.com/services",
+  session: "68c355e7-d255-4f4c-b912-d35b12f709c7"
+});
+
+let network, data;
+let unreacheableDevices = [];
+// Timer used for updating data
+let updateTimer;
+// 3 min
+let timeInterval = 180000;
+
+let statusMessage = {
+  warning_user_login:
+    "Please login into your Netatmo account to grant permissions to this app",
+  success_user_granted_netatmo_data_access: "Permission granted",
+  error_user_denied_netatmo_data_access:
+    "Permission to access Netatmo account data denied",
+  success_retrieve_wappsto_data: "Succesfully retrieved Wappsto data",
+  error_retrieve_wappsto_data: "Failed to retrieve Wappsto data"
+};
+
+wappsto.get(
+  "data",
+  {},
+  {
+    expand: 1,
+    subscribe: true,
+    success: function(collection) {
+      data = collection.first();
+
+      checkNetwork();
+    },
+    error: function(error) {
+      console.log(error);
+    }
+  }
+);
+
+// Check if network exists; if not then create it
+let checkNetwork = function() {
+  wappsto.get(
+    "network",
+    {},
+    {
+      expand: 5,
+      subscribe: true,
+      success: function(collection) {
+        if (collection.length > 0) {
+          network = collection.first();
+
+          setUpdateTimer();
+        } else {
+          network = createNetwork();
+          // get the users station data with which to populate the network
+          getStationData();
+        }
+      },
+      error: function(error) {
+        console.log(error);
+      }
+    }
+  );
+};
+
+// Update station data
+let updateStationData = function() {
+  unreacheableDevices = [];
+  // Remove previous unreacheable devices
+  updateWappstoData({ lostDevices: unreacheableDevices });
+
+  if (updateTimer) {
+    clearInterval(updateTimer);
+  }
+  // Destroy existing network and then create new network using new data
+  network
+    .destroy()

```

Comment:
do not delete the network, try to update it.
a patch/put request propagates to devices, values and states.
If you update /network/network_id and put in the body:
`
{
  "name": "test",
  "device": [{
    "meta":{
      "id": "device_id",
      "type": "device"
    },
    "name": "aaa"
  }]
}
`
if device_id does not exist under network_id, it will be created, updated otherwise